src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
             return;
         }
-        Entry<K, V> newEntry;
-        if (data.size() == cap) {
+        Entry<K, V> newEntry = new Entry<>(); 
+        if (data.size() >= cap) {
             newEntry = evict();
             data.remove(newEntry.getKey());
```

Comment:
**Scalability Issue**: Changing the condition from 'data.size() == cap' to 'data.size() >= cap' may lead to unnecessary eviction 
calls when the cache size is already below capacity. This could degrade the performance under high load, 
affecting scalability by performing redundant operations that could impact the throughput of cache operations. <br> **Fix**: Revert the condition to check for equality ('data.size() == cap') to avoid unnecessary evictions when the cache 
size is below the capacity. This ensures that evictions only occur when the cache is actually full, optimizing 
resource usage and maintaining high throughput under scale. <br> **Code Suggestion**: 
 ```
 -        if (data.size() >= cap) {
 +        if (data.size() == cap) {
 ```